tests/std/tests/P0220R1_searchers/test.cpp

```diff
@@ -290,6 +357,89 @@ void test_case_BM_unicode32() {
+    cout << searcher << " failed for needle \"" << needle << "\" and haystack \"" << haystack << "\".\n";
+    cout << "This is a randomized test.\n";
+    cout << "DO NOT IGNORE/RERUN THIS FAILURE.\n";
+    cout << "You must report it to the STL maintainers.\n";
+    assert(false);
+}
+
+void initialize_randomness(mt19937& mt) {
+    constexpr size_t n = mt19937::state_size;
+    constexpr size_t w = mt19937::word_size;
+    static_assert(w % 32 == 0);
+    constexpr size_t k = w / 32;
+
+    vector<uint32_t> vec(n * k);
+    random_device rd;
+    generate(vec.begin(), vec.end(), ref(rd));
+    seed_seq seq(vec.cbegin(), vec.cend());
+    mt.seed(seq);
+}
+
+void test_case_randomized_cases() {
+    const auto start = chrono::steady_clock::now();
+
+    mt19937 mt;
+    initialize_randomness(mt);
+
+    constexpr int Needles   = 150;
+    constexpr int Haystacks = 150;
+
+    constexpr size_t MaxNeedleLength = 10;
+    uniform_int_distribution<size_t> needle_length(1, MaxNeedleLength);
+    string needle(MaxNeedleLength, '?');
+    const auto needle_first = needle.c_str();
+
+    constexpr size_t MaxHaystackLength = 100;
+    uniform_int_distribution<size_t> haystack_length(1, MaxHaystackLength);
+    string haystack(MaxHaystackLength, '?');
+    const auto haystack_first = haystack.c_str();
+
+    for (int max_char = 'b'; max_char <= 'f'; ++max_char) {
+        uniform_int_distribution<int> characters('a', max_char);
+
+        for (int n = 0; n < Needles; ++n) {
+            needle.resize(needle_length(mt));
+            for (auto& ch : needle) {
+                ch = static_cast<char>(characters(mt));
+            }
+            const auto needle_last = needle_first + needle.size();
+
+            const default_searcher ds{needle_first, needle_last};
+            const boyer_moore_searcher bms{needle_first, needle_last};
+            const boyer_moore_horspool_searcher bmhs{needle_first, needle_last};
+
+            for (int h = 0; h < Haystacks; ++h) {
+                haystack.resize(haystack_length(mt));
+                for (auto& ch : haystack) {
+                    ch = static_cast<char>(characters(mt));
+                }
+                const auto haystack_last = haystack_first + haystack.size();
+
+                const auto correct = ds(haystack_first, haystack_last);
+
+                if (bms(haystack_first, haystack_last) != correct) {
+                    report_randomized_failure("boyer_moore_searcher", needle, haystack);
+                }
+
+                if (bmhs(haystack_first, haystack_last) != correct) {
+                    report_randomized_failure("boyer_moore_horspool_searcher", needle, haystack);
+                }
+            }
+        }
+    }
+
+    const auto finish = chrono::steady_clock::now();
+    const auto ms     = chrono::duration_cast<chrono::milliseconds>(finish - start);
+
+    if (ms.count() > 10'000) {

```

Comment:
If we named `ms` something descriptive like `elapsed`, we could stay in the duration domain here and test `elapsed > 10s` which is beautifully clear.

Reply:
Great idea, simplified!